.github/workflows/release.yaml

```diff
@@ -19,11 +19,104 @@ jobs:
         with:
           credentials: "${{ secrets.CHARMHUB_TOKEN }}" # FIXME: current token will expire in 2023-07-04
           github-token: "${{ secrets.GITHUB_TOKEN }}"
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run linters
+        run: tox -e lint
+
+  unit-test:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: python -m pip install tox
+      - name: Run tests
+        run: tox -e unit
+
+  integration-charm:
+    name: Integration tests for (lxd)
+    runs-on: ubuntu-latest
```

Comment:
As we're editing this, could we add a `needs` block to all these int-tests based on Jon's suggestion in MM?

```yaml
  integration-charm:
    name: Integration tests for charm (lxd)
    runs-on: ubuntu-latest
    needs:
      - lint
      - unit-test
      - lib-check
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Setup operator environment
        uses: charmed-kubernetes/actions-operator@main
        with:
          provider: lxd
      - name: Run integration tests
        run: tox -e integration-charm
```